app/src/main/java/com/android/sample/ui/activity/EditActivity.kt

```diff
@@ -518,47 +519,56 @@ fun EditActivityScreen(
                         startTime?.isNotEmpty() ?: false &&
                         duration.isNotEmpty(),
                 onClick = {
-                  if (!hourDateViewModel.isBeginGreaterThanEnd(
+                  // we don't want to allow creators to schedule start time of activities 1 hour
+                  // before they start
+                  if (dueDate.toDate().time - System.currentTimeMillis() <
+                      TimeUnit.HOURS.toMillis(1)) {
+                    Toast.makeText(
+                            context,
+                            "Activity must be scheduled at least 1 hour in advance",
+                            Toast.LENGTH_SHORT)
+                        .show()
```

Comment:
It is ok that this code is duplicated, because the class itself is duplicated. We shall fix that in another refactoring task.